src/app/hooks/useTasks.ts

```diff
@@ -1,58 +1,57 @@
 import { useState, useEffect } from "react";
-import { z } from "zod";
+import { Task } from "../types/task";
 
-const ZTaskSchema = z.object({
-  id: z.string(),
-  title: z.string(),
-  description: z.string(),
-  category: z.string(),
-  date: z.string(),
-});
-
-export type Task = z.infer<typeof ZTaskSchema>;
-export const useTasks = (userId: string | undefined) => {
+export function useTasks(userId: string | undefined) {
   const [tasks, setTasks] = useState<Task[]>([]);
 
   useEffect(() => {
-    if (!userId) return;
-    const storedTasks = localStorage.getItem(`tasks_${userId}`);
-    if (!storedTasks) {
-      setTasks([]);
-      return;
+    if (userId) {
+      fetchTasks();
     }
-    const parsedTask = ZTaskSchema.array().parse(JSON.parse(storedTasks));
-    setTasks(parsedTask);
   }, [userId]);
 
-  useEffect(() => {
-    if (!userId) return;
-    localStorage.setItem(`tasks_${userId}`, JSON.stringify(tasks));
-  }, [tasks, userId]);
-
-  const addTask = (task: Omit<Task, "id">) => {
-    const parsedTask = ZTaskSchema.omit({ id: true }).parse(task);
-    const newTask: Task = {
-      ...parsedTask,
-      id: crypto.randomUUID().toString(),
-    };
-    setTasks((prev) => [...prev, newTask]);
+  const fetchTasks = async () => {
+    try {
+      const response = await fetch("/api/listTasks");
+      if (!response.ok) {
+        throw new Error("Failed to fetch tasks");
+      }
+      const data = await response.json();
+      setTasks(data);
+    } catch (error) {
+      console.error("Failed to fetch tasks:", error);
+    }
   };
 
-  const deleteTask = (id: string) => {
-    setTasks((prev) => prev.filter((task) => task.id !== id));
+  const addTask = async (formData: FormData) => {
```

Comment:
**Suggestion: can you experiment with useing use-swr and use-mutate-swr for these requests?**

It saves you a little bit of boiler plate and can make things easier going forward